FlowCrypt/Controllers/Setup/SetupGenerateKeyViewController.swift

```diff
@@ -75,51 +77,113 @@ final class SetupGenerateKeyViewController: SetupCreatePassphraseAbstractViewCon
 
 extension SetupGenerateKeyViewController {
     private func setupAccountWithGeneratedKey(with passPhrase: String) {
-        Promise { [weak self] in
-            guard let self = self else { return }
-            self.showSpinner()
-
-            let userId = try self.getUserId()
-
-            try awaitPromise(self.validateAndConfirmNewPassPhraseOrReject(passPhrase: passPhrase))
-
-            let encryptedPrv = try self.core.generateKey(passphrase: passPhrase, variant: .curve25519, userIds: [userId])
-
-            try awaitPromise(self.backupService.backupToInbox(keys: [encryptedPrv.key], for: self.user))
-
-            let passPhrase = PassPhrase(value: passPhrase, fingerprints: encryptedPrv.key.fingerprints)
-
-            self.keyStorage.addKeys(keyDetails: [encryptedPrv.key], source: .generated, for: self.user.email)
-            self.passPhraseService.savePassPhrase(with: passPhrase, inStorage: self.shouldStorePassPhrase)
-
-            let updateKey = self.attester.updateKey(
-                email: userId.email,
-                pubkey: encryptedPrv.key.public,
-                token: self.storage.token
-            )
+        showSpinner()
+
+        cancellable = Just(passPhrase)
+            .subscribe(on: DispatchQueue.global())
```

Comment:
General idea is to control externally how to we want to run method. So method has no idea about queues and sync/async.